src/components/AutoComplete/AsyncAutoComplete.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { AsyncTypeahead } from 'react-bootstrap-typeahead';
+import PropTypes from 'prop-types';
+
+class AsyncAutoComplete extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      options: []
+    };
+
+    this.onSearchHandler = this.onSearchHandler.bind(this);
+  }
+
+  onSearchHandler = () => {
+    const options = this.props.onSearch();
```

Comment:
If this indeed Async options should come down as props along with `isLoading`.  As it is right now a user would not be able to something like
```jsx
handleSearch = (searchQuery) => {
  this.setState({ isLoading: true });
  fetch(`someUrl?query=${searchQuery}`).then(res => res.json()).then(res => {
    this.setState({ options: res.options, isLoading: false });
  })
}

render() { <AsyncAutoComplete onSearch={this.handleSearch} {...this.state} /> }
```

Reply:
As this component is still in progress, I have added already **isLoading** to the onSearch function.
Will upload it soon.